shell/platform/windows/win32_window.cc

```diff
@@ -153,13 +153,34 @@ Win32Window::MessageHandler(HWND hwnd,
         window->OnScroll(
             0.0, -(static_cast<short>(HIWORD(wparam)) / (double)WHEEL_DELTA));
         break;
+      case WM_UNICHAR: {
+        // Tell third-pary app, we can support Unicode.
+        if (wparam == UNICODE_NOCHAR)
+          return TRUE;
+        // DefWindowProc will send WM_CHAR for this WM_UNICHAR.
+        break;
+      }
       case WM_CHAR:
-      case WM_SYSCHAR:
-      case WM_UNICHAR:
-        if (wparam != VK_BACK) {
-          window->OnChar(static_cast<unsigned int>(wparam));
+      case WM_SYSCHAR: {
+        if (wparam == VK_BACK)
+          break;
+        char32_t ch = static_cast<char32_t>(wparam);
+        // What we expect to pass in is a Unicode, if it's UTF-32, it doesn't
+        // need any processing, and if it's UTF-16, it's converted to Unicode.
+        static char32_t lead_ch = 0;
```

Comment:
Same note here about naming. Maybe call this `lead_surrogate`

Reply:
done.